test/storage/online_file_source.test.cpp

```diff
@@ -425,3 +425,40 @@ TEST(OnlineFileSource, ChangeAPIBaseURL){
     fs.setAPIBaseURL(customURL);
     EXPECT_EQ(customURL, fs.getAPIBaseURL());
 }
+
+
+TEST(OnlineFileSource, TEST_REQUIRES_SERVER(LowHighPriorityRequests)) {
+    util::RunLoop loop;
+    OnlineFileSource fs;
+    std::size_t response_counter = 0;
+    std::size_t NUM_REQUESTS = 3;
+
+    NetworkStatus::Set(NetworkStatus::Status::Offline);
+
+    // requesting a low priority resource
+    Resource low_prio{ Resource::Unknown, "http://127.0.0.1:3000/test" };
+    low_prio.setLowPriority();
+    std::unique_ptr<AsyncRequest> req_0 = fs.request(low_prio, [&](Response) {
+        response_counter++;
+        req_0.reset();
+        EXPECT_EQ(response_counter, NUM_REQUESTS); // make sure this is responded last
```

Comment:
As you brought up on chat @chaupow, this test doesn't seem to be correctly written. I think the low priority request being requested first results in it being satisfied first. To correctly test the prioritization, you either want to test that a flood of low priority requests doesn't take priority over the regular ones, OR that a flood of regular requests takes priority over a few pending low priority requests. 
Even with `maximumConcurrentRequests=1` for this type of test, you will need a couple more requests in order to create a reliable test. 
You might also want to try different request URLs in case the platform implementations on the `OnlineFileSo urce` have caching that affects the response timing.